### ISSUE TITLE
Improve documentation of the crate

### ISSUE THREAD

**Issue by DDtKey** [testcontainers/testcontainers-rs-modules-community#2]
We need to enrich the crate with documentation

Depends on #1

```[tasklist]
### Tasks
- [x] Contribution guide - @ubiratansoares is assigned
- [x] Improve crate-level documentation (docs.rs)
- [ ] Document usage of each existing module
- [x] Align readme with documentation
- [x] New issue template
- [x] Activate `#![deny(missing_docs)]` at crate level to avoid missing documentation
```

**Comment by DDtKey** [comment]
Two remaining tasks are:
- Document usage of each existing module
We have a number of modules which has been moved from `testcontainers` repo, most of them don't have a documentation. Need to actualize and write a documentation like this: https://github.com/testcontainers/testcontainers-rs-modules-community/blob/6a4d10acbedd19f54a505afd09b66343744b4a64/src/rabbitmq/mod.rs#L6-L29

- Contribution guide

I would be glad for any help

**Comment by level-one-learner** [comment]
Okay I'll have a go at adding documentation for  the use of each module and submit a PR.

**Comment by ubiratansoares** [comment]
@DDtKey Is there anyone assigned to Contribution Guidelines?  If not, I'd love helping with that! 🙂

**Comment by DDtKey** [comment]
No, feel free to take on this task!
Thank you!🙏

**Comment by DDtKey** [comment]
To actualize the state:
We still need to cover undocumented modules. But there is no need to document all of them at once.

I'm happy to accept PRs with documentation for 1 module at a time.

**Comment by CommanderStorm** [comment]
to update the issue:
Most items now (assuming the tow PRs get merged) have doc-strings, except the following, where I don't quite know how to document this.

https://github.com/testcontainers/testcontainers-rs-modules-community/blob/1886aa12b343ff02a55418e2d269d2d8d166c864/src/consul/mod.rs#L31-L33

**Comment by CommanderStorm** [comment]
@ruslansennov since you contributed this:

What would be a good doc-string for this aspect? I don't know enough about said docker container to document this

**Comment by CommanderStorm** [comment]
🎉

**Comment by CommanderStorm** [comment]
Thank you people for doing this initiative in the first place.
Testcontainers make testing much nicer ^^
